### ISSUE TITLE
CachedNetworkImage crashes when it is not in SizedBox and width, height is null

### ISSUE THREAD

**Issue by benadrasni** [renefloor/flutter_cached_network_image#128]
I put statement below inside your example _testContent() and got en error. Version 0.5.1 is working without specifying width and height (e.g. inside Card):

CachedNetworkImage(
   imageUrl: "http://via.placeholder.com/350x200",
),

Error:
I/flutter (29008): The following assertion was thrown during performLayout():
I/flutter (29008): BoxConstraints forces an infinite height.
I/flutter (29008): These invalid constraints were provided to RenderSemanticsAnnotations's layout() function by the
I/flutter (29008): following function, which probably computed the invalid constraints in question:
I/flutter (29008):   RenderConstrainedBox.performLayout (package:flutter/src/rendering/proxy_box.dart:258:13)
I/flutter (29008): The offending constraints were:
I/flutter (29008):   BoxConstraints(w=411.4, h=Infinity)

**Comment by renefloor** [comment]
Hmm good point. That is because it now uses a stack to put the wigets on top of each other instead of showing one after the other. I'll think of how to fix this nicely.

**Comment by panuavakul** [comment]
This is probably not the best solution, but I had to do something asap so I wrapped the CachedNetworkImage in `IntrinsicHeight` and `IntrinsicWidth`.

``` dart
IntrinsicHeight(
  child: IntrinsicWidth(
    child: CachedNetworkImage()
  )
)
```
for it to work.

**Comment by panuavakul** [comment]
@renefloor Pardon me, if I don't know what am I talking about. I was doing some debugging as one of my projects heavily relies on `cached_network_image`.

I'm not sure if the `?? double.inifinity` [here](https://github.com/renefloor/flutter_cached_network_image/blob/master/lib/src/cached_image_widget.dart#L285) is used for anything?
```
height: widget.height ?? double.infinity,
width: widget.width ?? double.infinity,
```
 In my understanding (and I could be really wrong) the value of widget.height there should be `null` most of the time causing the container to have `double.infinity` height and width.

I tried removing it on my fork [here](https://github.com/panuavakul/flutter_cached_network_image/blob/bugfix/fix_inifinite_layout/lib/src/cached_image_widget.dart#L285), and and pass the null to the Container and let it layout itself, and it seems to work well at least for my use cases (inside a ListView, Stack, and other widgets that does not have constraints on child size).

So maybe even the line 
```
height: widget.height,
width: widget.width,
```
can be removed all together. But by doing this, I'm not too sure if there will be any other problem elsewhere.

**Comment by renefloor** [comment]
I had added that container because otherwise Boxfit didn't work automatically. The widget just becomes smaller untill it fits. However, I didn't think about this side effect. I'll revert that commit. If people need the Boxfit to work they just have to add width and/or height.

**Comment by renefloor** [comment]
published as 0.6.1

**Comment by AdamBCo** [comment]
@renefloor I am experiencing this issue at the moment. What was the fix exactly?
